### PR TITLE
VFB-24 View Client Page

### PR DESCRIPTION
**Pull request by SamYap-Softwire**
`/clients/[id]` now displays the client information with the matching primary key

- [x] Make sure you've verified it works via `npm run dev`
- [x] Make sure you've linted via `npm run lint`
    - can run `npm run lint_fix` to try and fix as any mistakes automatically as possible!
- [x] Make sure you've tested via `npm run test`

### CODE REVIEW

File: src/pdf/ShoppingList/dataPreparation.ts
Comment (TomCusack-Softwire) [inline src/pdf/ShoppingList/dataPreparation.ts:1]: probably should move this out of /pdf/shoppinglist if other files are going to start using it! maybe a thing to do in the week after the demo?

File: src/app/clients/[id]/ViewClientPage.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+const KeyText = styled.p`
+    font-weight: bold;
+`;
+
+const ViewClientPage: React.FC<Props> = ({ clientID, clientData }) => {
+    return (
+        <CenterComponent>
+            <StyledCard>
+                <TempDiv>
```
Comment (TomCusack-Softwire) [inline src/app/clients/[id]/ViewClientPage.tsx:38]: does both inner and outer need to be wrapped in TempDivs?

File: src/app/clients/[id]/getClientData.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    prepareClientSummary,
+    prepareHouseholdSummary,
+    prepareRequirementSummary,
+    RequirementSummary,
+} from "@/pdf/ShoppingList/dataPreparation";
+import { fetchClients, fetchFamilies } from "@/pdf/ShoppingList/databaseFetch";
+
+export type ClientData = ClientSummary & HouseholdSummary & RequirementSummary;
+
+const getClientData = async (clientID: string): Promise<ClientData> => {
```
Comment (TomCusack-Softwire) [inline src/app/clients/[id]/getClientData.ts:13]: 😍

File: src/app/clients/[id]/getClientData.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import {
+    ClientSummary,
+    HouseholdSummary,
+    prepareClientSummary,
+    prepareHouseholdSummary,
+    prepareRequirementSummary,
+    RequirementSummary,
+} from "@/pdf/ShoppingList/dataPreparation";
+import { fetchClients, fetchFamilies } from "@/pdf/ShoppingList/databaseFetch";
```
Comment (TomCusack-Softwire) [inline src/app/clients/[id]/getClientData.ts:9]: may want to be careful - these functions assume client-side (supabaseClient is used) but I think these are currently server-side, which may cause some weird bugs

**Review by TomCusack-Softwire — APPROVED** [review]

File: src/app/clients/[id]/ViewClientPage.tsx (hunk 1 above)
Comment (SamYap-Softwire) [reply src/app/clients/[id]/ViewClientPage.tsx:38]: thanks for catching this TT 
neither of them are needed - hence the `Temp` in the name ^^

File: src/pdf/ShoppingList/dataPreparation.ts
Comment (SamYap-Softwire) [reply src/pdf/ShoppingList/dataPreparation.ts:1]: yep agree, i was thinking of doing this but wasn't sure where I should put it in - will do after demo though!

File: src/app/clients/[id]/getClientData.ts (hunk 3 above)
Comment (SamYap-Softwire) [reply src/app/clients/[id]/getClientData.ts:9]: temporary fix now but will do a proper fix (separating client and server) after the demo!

**Pull request merged**